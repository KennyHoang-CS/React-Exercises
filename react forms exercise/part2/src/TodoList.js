import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = () => {
    const [tasks, setTasks] = useState([]);
    const addTask = (newTask) => {
        setTasks([...tasks, {...newTask }]);
    }
    const deleteTask = (taskName) => {
        setTasks(tasks.filter(task => task.task != taskName));
    }
    return (
        <div>
            <h3>Todo List</h3>
            <NewTodoForm addTask={ addTask }/>
            {tasks.map(({ task }) => <Todo key={task} task={task} deleteTask={deleteTask} />)}
        </div>
    )
}

export default TodoList;