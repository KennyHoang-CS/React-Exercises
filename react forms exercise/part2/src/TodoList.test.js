import { render, fireEvent, waitFor } from '@testing-library/react';
import TodoList from './TodoList';

function addTask(taskList, task='myRandomTask') {
    const taskInput = taskList.getByLabelText('New Task');
    fireEvent.change(taskInput, { target: { value: task } });
    const addButton = taskList.getByText('Add new task!');
    fireEvent.click(addButton);
}

it('TodoList should render', () => {
    render(<TodoList/>);
});

it('TodoList matches snapshot', () => {
    const { asFragment } = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});

it('works: adds a new task', () => {
    const taskList = render(<TodoList/>);
    addTask(taskList);  // adding 'myRandomTask' to the TodoList. 

    // our task listed as 'myRandomTask' should be on the document. 
    expect(taskList.getByText('myRandomTask')).toBeInTheDocument();
});

it('works: removes a task', () => {
    const taskList = render(<TodoList/>);
    addTask(taskList);  // adding 'myRandomTask' to the TodoList. 

    const delButton = taskList.getByText('X');
    // our task listed as 'myRandomTask' should be on the document. 
    expect(taskList.getByText('myRandomTask')).toBeInTheDocument();

    // delete the 'myRandomTask' now. 
    fireEvent.click(delButton);
    expect(taskList.queryByText('myRandomTask')).not.toBeInTheDocument();
});