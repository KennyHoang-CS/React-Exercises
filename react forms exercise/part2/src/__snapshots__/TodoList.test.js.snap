// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="task"
      >
        New Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="new task"
        type="text"
        value=""
      />
      <button>
        Add new task!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
